File: src/cpp/Herbivore.cpp
/*
 * File:   Herbivore.cpp
 * Author: David KALONJI
 * Email : kalonjidvd@gmail.com
 *
 * Created on 26 aug 2016, 13:49
 * Last modified on Apr 20 16:53 2017
 */

/**
 Implementation du type concret Herbivore de regimeAlimentaire.
 */

#include "Herbivore.h"

Herbivore::Herbivore(){} // Constructeur

void Herbivore::seNourrir(Faune* consommateur, ObjetJungle* consommable)
{
    Ressource* laPlante = dynamic_cast<Ressource*>(consommable);

    while(consommateur->getEnergie() <= 71)
    {
        if(laPlante->getQuantite() > 29)
        {
            laPlante->diminuerDeQuantite(30);
            consommateur->augmenterDEnergie(30);
        }
    }
}

Herbivore::~Herbivore(){} // Destructeur

File: src/java/Herbivore.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * File:   Herbivore.java
 * Author: David KALONJI
 * Email : kalonjidvd@gmail.com
 *
 * Created on 27 août 2016, 13:49
 * Last modified on Apr 21 16:41 2017
 */

/**
 * Classe concrete Herbivore implementant l'interface RegimeAlimentaire.
 */
public class Herbivore implements RegimeAlimentaire{

    /**
     * Implementation de la methode abstraite de l'interface RegimeAlimentaire.
     *
     * @param consommateur
     * @param laPlante
     */
    public void seNourrir(Faune consommateur, ObjetJungle laPlante){
        while(consommateur.getEnergie() <= 71)
        {
            if(((Arbre)laPlante).getQuantite() > 29)
            {
                ((Arbre)laPlante).diminuerDeQuantite(30);
                consommateur.augmenterDEnergie(30);
            }
        }
    }
}
